code/modules/heavy_vehicle/components/_components.dm

```diff
@@ -47,8 +47,9 @@
 	return
 
 /obj/item/mech_component/proc/return_diagnostics(var/mob/user)
-	to_chat(user, SPAN_NOTICE("[capitalize_first_letters(src.name)]:"))
-	to_chat(user, SPAN_NOTICE(" - Integrity: <b>[round(((max_damage - total_damage) / max_damage) * 100, 0.1)]%</b>" ))
+	. = list()
+	. += SPAN_NOTICE("[capitalize_first_letters(src.name)]:")
+	. += SPAN_NOTICE(" - Integrity: <b>[round(((max_damage - total_damage) / max_damage) * 100, 0.1)]%</b>" )
```

Comment:
I do not think this is going to work.
-> Afaik this is being called by `/mob/living/heavy_vehicle/attackby()` in the mech_interaction.dm
-> It expects that the diagnostic data is printed to chat and not returned as a list (unless you have updated htat?